src/text.ts

```diff
@@ -295,3 +295,28 @@ export function extractIDfromSlug(slug: string): string | undefined {
   }
   return slug.split("-").pop()
 }
+
+/**
+ * Abbreviate number
+ * @function
+ * @param number The number to be abbreviated
+ * @returns The string representation of the abbreviated number
+ */
+export const abbreviateNumber = (number: number | undefined | null): string => {
+  const abbreviations = [
+    { value: 1e9, symbol: "B" },
+    { value: 1e6, symbol: "M" },
+    { value: 1e3, symbol: "k" },
+  ]
+
+  if (!number) return "n/a"
```

Comment:
should it not return null or throw Error instead ?

Reply:
@king-d-dev would prefere to throw null

Reply:
Addressed